### PR TITLE
Clicking on the cancel button on a newly created todo item will now d…

### PR DESCRIPTION
**Pull request by giedrius-timinskis**
…elete it

### CODE REVIEW

**Comment by mrmckeb** [comment]
Do you want to make this a PR into develop?

File: src/client/views/components/itemList.jsx [hunk 1]
```diff
@@ -38,7 +38,8 @@ export default React.createClass({
                     onDelete={this.props.onDelete}
                     onEdit={this.props.handleEdit}
                     onUpdate={this.props.onUpdate}
-                    title={title} />
+                    title={title}
+                    addingNewItem={addingNewItem} />
```
Comment (mrmckeb) [inline src/client/views/components/itemList.jsx:42]: Good approach. I'd suggest 'isNewItem' for the name though (throughout), what do you think?

File: src/client/views/components/todoItem.jsx [hunk 2]
```diff
@@ -75,12 +75,16 @@ export default React.createClass({
...
+        const {description, dueDate, title, addingNewItem} = this.props;
+        if (addingNewItem) {
+            this.handleDelete();
+        } else {
+            this.setState({
+                inEdit: false,
+                description,
+                dueDate: dueDate.toDateString(),
+                title});
+        }
```
Comment (mrmckeb) [inline src/client/views/components/todoItem.jsx:87]: This is good. Another way you could approach this is:

```
if (addingNewItem) { 
    return this.handleDelete();
}
this.setState({ ... });
```

Then you don't need the else, and it keeps the code a little shorter. But it's also a little less _explicit_.

Comment (giedrius-timinskis) [inline src/client/views/components/todoItem.jsx:87]: I think that when skimming through the code it's easier to miss the `return` statement and misinterpret the code, while the `else` here makes the intent more clear. I'm inclined to leave this as it is.

Comment (mrmckeb) [inline src/client/views/components/todoItem.jsx:87]: Are you so _inclined_? 😛 I'm happy to leave that.

**Pull request closed without merging**